src/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -4,21 +4,27 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
+using System.Diagnostics;
 using System.Linq.Expressions;
 using System.Reflection;
 
 namespace System.Linq
 {
-    internal class EnumerableRewriter : OldExpressionVisitor
+    internal class EnumerableRewriter : ExpressionVisitor
```

Comment:
How often / in what situations are EnumerableRewriters constructed?  This is adding two dictionaries to every construction.